app/models/miq_report.rb

```diff
@@ -83,6 +83,18 @@ def self.having_report_results(options = {})
     q
   end
 
+  def add_schedule(data)
```

Comment:
This method changes the content/structure of the data that's passed in. WDYT about either `dup`ing or creating a new hash to pass to `create!`?